src/solver/impls/arkode/arkode.cxx

```diff
@@ -195,30 +195,29 @@ int ArkodeSolver::init() {
   // Put the variables into uvec
   save_vars(N_VGetArrayPointer(uvec));
 
-  ASSERT1(solve_explicit or solve_implicit);
+  ASSERT1(imex or solve_explicit or solve_implicit);
 
-  const auto& explicit_rhs = [this]() {
-    if (imex) {
-      return arkode_rhs_explicit;
-    } else {
-      return solve_explicit ? arkode_rhs : nullptr;
-    }
-  }();
-  const auto& implicit_rhs = [this]() {
-    if (imex) {
-      return arkode_rhs_implicit;
-    } else {
-      return solve_implicit ? arkode_rhs : nullptr;
-    }
-  }();
-
-  arkode_mem = callWithSUNContext(ARKStepCreate, suncontext, explicit_rhs, implicit_rhs,
-                                  simtime, uvec);
+  if(imex or (solve_explicit and solve_implicit))
+  {
+    imex = true;
+    arkode_mem = callWithSUNContext(ARKStepCreate, suncontext, arkode_rhs_explicit, arkode_rhs_implicit,
+                                    simtime, uvec);    
+  }
+  else if (solve_explicit)
+  {
+    arkode_mem = callWithSUNContext(ARKStepCreate, suncontext, arkode_rhs, nullptr,
+                                    simtime, uvec);    
+  }
+  else
+  {
+    arkode_mem = callWithSUNContext(ARKStepCreate, suncontext, nullptr, arkode_rhs,
+                                    simtime, uvec);
+  }
```

Comment:
```suggestion
  else if (solve_explicit && !imex)
  {
    arkode_mem = callWithSUNContext(ARKStepCreate, suncontext, arkode_rhs, nullptr,
                                    simtime, uvec);    
  }
  else if (solve_implicit && !imex)
  {
    arkode_mem = callWithSUNContext(ARKStepCreate, suncontext, nullptr, arkode_rhs,
                                    simtime, uvec);
  } else {
    throw BoutException("Incompatible imex, explicit, and implicit options\n");
  }
```
The assert above can be removed with this

Reply:
ASSERT1 is removed